tests/containers/rootless_podman.pm

```diff
@@ -0,0 +1,65 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Test rootless mode on podman.
+# - add a user on the /etc/subuid and /etc/subgid to allow automatically allocation subuid and subgid ranges.
+# - check uids allocated to user (inside the container are mapped on the host)
+# - give read access to the SUSE Customer Center credentials to call zypper from in the container
+#   proper way to add Access Control List is via `setfacl` which is not available so we just do
+#   `chmod` instead!! This grants the current user the required access rights
+# - Test rootless container:
+#   * container is launched with default root user
+#   * container is launched with existing user id
+#   * container is launched with keep-id of the user who run the container
+# - Restore /etc/zypp/credentials.d/ credentials
+# Maintainer: qa-c team <qa-c@suse.de>
+
+use Mojo::Base qw(consoletest);
+use testapi;
+use utils;
+use containers::common;
+use containers::container_images;
+use suse_container_urls 'get_suse_container_urls';
+use version_utils qw(get_os_release);
+
+sub run {
+    my ($self) = @_;
+    my ($image_names, $stable_names) = get_suse_container_urls();
+    my ($running_version, $sp, $host_distri) = get_os_release;
+    my $runtime = "podman";
+
+    install_podman_when_needed($host_distri);
+    allow_selected_insecure_registries(runtime => $runtime);
+    my $user      = $testapi::username;
```

Comment:
Can you please make a subroutine out of this block? It can stay in this file.

Reply:
which block?

Reply:
I think @pdostal asking you the same as I already did :) but you answered that it does not make sense https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/11878#discussion_r576208803 ;)

Reply:
it was in a different file tho. but i guess my answer is the same.

Reply:
```perl
sub prepare_rootless_podman {
  my $user      = $testapi::username;
  my $check_msg = 'Checking allocation range of user';

  script_run "grep $user /etc/subuid || echo /etc/subuid has no uid range for $user", output => $check_msg;
  script_run "grep $user /etc/subgid || echo /etc/subgid has no gid range for $user", output => $check_msg;
  assert_script_run "usermod --add-subuids 200000-201000 --add-subgids 200000-201000 $user";
  assert_script_run "grep $user /etc/subuid", fail_message => "subuid range not assigned for $user";
  assert_script_run "grep $user /etc/subgid", fail_message => "subgid range not assigned for $user";

  # Workaround instead of
  # "setfacl -m u:$user:r /etc/zypp/credentials.d/*"
  assert_script_run "chmod -R 666 /etc/zypp/credentials.d/*";
}
```

Reply:
Wondering if it would not be enough to use `podman $whatever_cmd --uidmap ... ` instead of doing system wide changes

Reply:
i just followed the documentation. `--uidmap` should work but i havent tried. Also we could revert the change in the end of the module but i cant see what harm causes if not.

Reply:
tried and failed. unless there is something i am missing the `usermod` is mandatory. and at least it is as the documentation states it